### ISSUE TITLE
Multiple `__name__` matchers not permitted

### ISSUE THREAD

**Issue by gerboland** [GreptimeTeam/promql-parser#70]
A PromQL query can contain multiple `__name__` matchers, e.g.
```
{__name__=~"node_cpu_.+", __name__=~".+total"}
```
[Here's a demo of such a query](http://demo.robustperception.io:9090/graph?g0.expr=%7B__name__%3D~%22node_cpu_.%2B%22%2C%20__name__%3D~%22.%2Btotal%22%7D&g0.tab=1&g0.stacked=0&g0.range_input=1h)

However with this lib it fails with `metric name must not be set twice: '.+total' or 'node_cpu_.+'`

Many thanks for this high quality effort btw!

**Comment by yuanbohan** [comment]
Thanks for this issue, and I will fix it ASAP, and related test cases will be added.

**Comment by yuanbohan** [comment]
Hi @gerboland , this bug has been fixed in v0.1.3.

```
r#"{__name__=~"http.*",__name__=~".*total",environment=~"staging|testing|development",method!="GET"} @ 1609746000 offset 5m"#
```

The AST is:

```
AST: VectorSelector(VectorSelector { name: None, matchers: Matchers { matchers: {Matcher { op: Re(http.*), name: "__name__", value: "http.*" }, Matcher { op: NotEqual, name: "method", value: "GET" }, Matcher { op: Re(staging|testing|development), name: "environment", value: "staging|testing|development" }, Matcher { op: Re(.*total), name: "__name__", value: ".*total" }} }, offset: Some(Pos(300s)), at: Some(At(SystemTime { tv_sec: 1609746000, tv_nsec: 0 })) })
```

**Comment by gerboland** [comment]
@yuanbohan fantastic, thank you
